### PR TITLE
add DISCO_L496AG + STModCellular BG96 support

### PR DESCRIPTION
**Pull request by soleilplanet**
### Summary of changes 

[X] I confirm this contribution is my own and I agree to license it with Apache 2.0.

[X] I confirm the moderators may change the PR before merging it in.

For new board enablements only:

[X] I confirm the board is [Mbed Enabled](https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/) and passes the Mbed Enabled test set.

[X] I confirm the contribution has been tested properly and the tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.

[X] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes or the content in folders `mbed-os` and `mbed-cloud-client` were not modified in order to pass the tests.

- Maintainers of this repository update the Mbed OS and Client hashes.
- Please report any issues through issues in relevant repositories.

Add support for DISCO_L496AG + STModCellular BG96.
Integration tests all passed, Pelion E2E tests all passed except for update test due to python package issue. Update campaign succeeded with manual 'mbed dm' commands.
[disco_l496ag_external_kvstore_pelion_e2e_test.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4196185/disco_l496ag_external_kvstore_pelion_e2e_test.log)
[disco_l496ag_external_kvstore_integration_test.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4196186/disco_l496ag_external_kvstore_integration_test.log)

### CODE REVIEW

**Comment by iot-pdmc** [comment]
Thanks for your changes.
@ARMmbed/pdmc-integrators please review.

**Comment by teetak01** [comment]
@soleilplanet does this not depend on https://github.com/ARMmbed/mbed-os/pull/12411 ?

**Comment by soleilplanet** [comment]
@teetak01 yes, this depends on https://github.com/ARMmbed/mbed-os/pull/12411, just forgot to put this information. Sorry about that.
I managed to get my python environment fixed, but the update test failed because it couldn't complete the the campaign within the specified timeout. (the default timeout is too short for NB-IoT network; my best record is 8 minutes, but it could take 30 minutes or longer, which depends on signal quality )

**Comment by teetak01** [comment]
Thanks for clarifying. I have already mentioned earlier for the testing team about this issue with the test-timeout.

**Comment by soleilplanet** [comment]
@teetak01 I've added an additional line and it is now not dependent on the Mbed-OS PR.

File: mbed_app.json [hunk 1]
```diff
@@ -266,6 +266,47 @@
...
+        "DISCO_L496AG": {
+            "target.macros_add": [
+                "CELLULAR_DEVICE=STModCellular"
+            ],
+            "target.network-default-interface-type"     : "CELLULAR",
+            "target.bootloader_img"                     : "bootloader/mbed-bootloader-disco_l496ag-external_qspif-kvstore-4.1.2.bin",
+            "target.header_offset"                      : "0x11000",
+            "target.app_offset"                         : "0x11400",
+            "target.components_add"                     : ["FLASHIAP"],
+            "target.components_add"                     : ["QSPIF", "LWIP", "STMOD_CELLULAR"],
```
Comment (teetak01) [inline mbed_app.json]: This configuration will not work. There cannot be duplicate configurations.

**Comment by soleilplanet** [comment]
@teetak01 here are the logs with the new commit.
[pelion_e2e_test_updated.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4198111/pelion_e2e_test_updated.log)
[l496ag_stmodcellular_bg96_mbed_os_integration_test.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4198114/l496ag_stmodcellular_bg96_mbed_os_integration_test.log)

**Comment by teetak01** [comment]
Can you fix the configuration? I am not sure that will even compile? Having two identical entries in .json should not even pass Mbed CLI verification, or if it does, one of those parameters would get silently ignored.

**Comment by soleilplanet** [comment]
@teetak01  it would compile and here's the log with a dump of the mbed_app.json:
[show_config_and_compile.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4198175/show_config_and_compile.log)
I'm basically running a script to call:
    **cat mbed_app.json | grep -A40 'DISCO_L496AG'
    mbed compile**
combining those two lines in one will not be able to build; you will see 'DeviceKey was not declared' during compilation. That's why I filed the PR in Mbed-OS at the first place.
(and I guess I would need to file an issue to the tools team)

**Comment by soleilplanet** [comment]
@teetak01 I found that the build error I mentioned above is not observed after I upgraded python to 2.7.17 (was 2.7.12) with single line of 'components_add'
So I'm going to merge those two lines and test again.

File: mbed_app.json (hunk 1 above)
Comment (teetak01) [reply mbed_app.json]:
```suggestion
-            "target.components_add"                     : ["FLASHIAP"],
-            "target.components_add"                     : ["QSPIF", "LWIP", "STMOD_CELLULAR"],
+            "target.components_add"                     : ["QSPIF", "LWIP", "STMOD_CELLULAR", "FLASHIAP"],

```

**Comment by teetak01** [comment]
Yeah good. I was not able to reproduce the issue either.

**Comment by teetak01** [comment]
You could also rebase the PR.

**Comment by soleilplanet** [comment]
test logs with combined components_add:
[integration_test_log_2.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4199130/integration_test_log_2.log)
[pelion_e2e_test_2.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4199131/pelion_e2e_test_2.log)

**Comment by teetak01** [comment]
build-only

**Comment by iot-pdmc** [comment]
# [Test run: FAILED](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/600/)
***

**Failed to build following targets:**
DISCO_L496AG-ARM-TCP
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/600)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Comment by soleilplanet** [comment]
updated test log with new commit:
[pelion_e2e_test_3.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4200715/pelion_e2e_test_3.log)
[integration_test_log_3.log](https://github.com/ARMmbed/mbed-os-example-pelion/files/4200716/integration_test_log_3.log)

**Comment by teetak01** [comment]
build-only

**Comment by iot-pdmc** [comment]
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/601/)
***
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/601)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Comment by teetak01** [comment]
@soleilplanet this would still require:

1) Add README entry for this board.
2) Squash the commits together
3) Rebase locally with master.

**Comment by soleilplanet** [comment]
@teetak01 I just squashed and rebased from master, but I'm not sure if that's the result you'd like to see. Would you please check? 
Thanks!

**Comment by teetak01** [comment]
build-only

**Comment by iot-pdmc** [comment]
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/602/)
***
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/602)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Comment by teetak01** [comment]
Looks good, thanks @soleilplanet

**Pull request merged**